### PR TITLE
Staff-Related Insights 

### LINKED ISSUE #178: Insight for Questions Helped by TA, Time
**Description**

A new insight that breaks down the number of questions helped by each of the course's staff, broken down by time + TA.

**Success Criterias**

Please provide the detailed requirements for this feature:

-   A new visual which displays the total number of questions helped over time, with options/existing subvisuals breaking it down by TA.

**Context**

Please provide any relevant information about this feature:

-   None at the moment

### PR DESCRIPTION
**Pull request by bhunt02**
# Description

* Added an insight showing the average wait/help/total time for staff members of a course
* Added an insight with Gantt charts showing each weekday's question's helped by each staff member
* Added an insight showing the total number of questions, async questions helped by staff members
* Added new insight type "MultipleGanttChart" which allows for the grouping of multiple Gantt chart outputs in one insight component, made corresponding changes to display logic
* Fixed an encroaching issue where lack of retrieved data in the Help Seeking insight would cause an overload to memory due to data being padded from the epoch to the present day, as no minimum day could be identified (particularly a problem with the Week timespan option, where empty data points are populated for each day rather than each month
* Fixed an issue with chart legends overflowing when too many unique data keys are in use

Closes #178 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests)
Provide instructions so we can reproduce. 

- [x] Insight appearance tested with the production data dump to ensure things lined up
- [x] Integration & unit tests for previous implementation were updated

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any work that this PR is dependent on has been merged into the main branch
- [x] Any UI changes have been checked to work on desktop, tablet, and mobile

### CODE REVIEW

**Review by AdamFipke — COMMENTED** [review]
Realized I made this comment and it never sent lol

File: packages/frontend/app/(dashboard)/course/[cid]/(insights)/insights/staff/page.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import InsightsPage from '@/app/(dashboard)/course/[cid]/(insights)/insights/page'
+export default InsightsPage
```
Comment (AdamFipke) [inline packages/frontend/app/(dashboard)/course/[cid]/(insights)/insights/staff/page.tsx:2]: I found one small thing, there seems to be a missing 3:30pm tick between 1pm and 6pm?

image

Though i'd say only bother fixing this if it's easy to do so

Comment (bhunt02) [reply packages/frontend/app/(dashboard)/course/[cid]/(insights)/insights/staff/page.tsx:2]: i'm pretty sure this is just an unfortunate side effect of recharts and avoiding putting ticks where they won't fit

File: packages/frontend/app/(dashboard)/course/[cid]/insights/context/InsightsContext.tsx [hunk 2]
```diff
@@ -52,6 +64,14 @@ const InsightContextProvider: React.FC<{
     })()
   }, [courseId, page, search])
 
+  useEffect(() => {
+    ;(async () => {
```
[the remark this answers is not in the document]
Comment (bhunt02) [reply packages/frontend/app/(dashboard)/course/[cid]/insights/context/InsightsContext.tsx:68]: thankfully it's not gonna have that effect due to being in a {} block, but wasn't this in a previous PR and I tried to remove it? probably the linter :(

**Review by AdamFipke — APPROVED** [review]
Looks good! Nice work

**Pull request merged**